### PR TITLE
[Qt] attribute replaceable (RBF) transactions

### PR DESCRIPTION
**Pull request by jonasschnelli**
This attributes transactiontable entries (one or more per wtx) if the transactions signals opt-in-RBF.
A light orange color together with a new icon (for "incoming" and "outgoing" transactions).

bildschirmfoto 2016-04-05 um 16 05 34

### CODE REVIEW

File: src/qt/transactionrecord.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include "base58.h"
 #include "consensus/consensus.h"
 #include "main.h"
+#include "policy/rbf.h"
```
Comment (paveljanik) [inline src/qt/transactionrecord.cpp]: Do you need this here?

**Comment by MarcoFalke** [comment]
Concept ACK

File: src/qt/transactiondesc.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ QString TransactionDesc::FormatTxStatus(const CWalletTx& wtx)
         else if (GetAdjustedTime() - wtx.nTimeReceived > 2 * 60 && wtx.GetRequestCount() == 0)
             return tr("%1/offline").arg(nDepth);
         else if (nDepth == 0)
-            return tr("0/unconfirmed, %1").arg((wtx.InMempool() ? tr("in memory pool") : tr("not in memory pool"))) + (wtx.isAbandoned() ? ", "+tr("abandoned") : "");
+            return tr("0/unconfirmed, %1").arg((wtx.InMempool() ? tr("in memory pool") : tr("not in memory pool"))) + (wtx.signalsOptInRBF() ? ", "+tr("replaceable") : "") + (wtx.isAbandoned() ? ", "+tr("abandoned") : "");
```
Comment (paveljanik) [inline src/qt/transactiondesc.cpp]: We will present transactions signalling opt-in RBF as "replaceable". Users will ask ask, what it means.

**Comment by paveljanik** [comment]
Concept ACK

File: src/qt/transactionrecord.cpp (hunk 1 above)
Comment (jonasschnelli) [inline src/qt/transactionrecord.cpp]: Right. This is not required. Fixed.

**Comment by btcdrak** [comment]
Travis needs a restart now.

**Comment by petertodd** [comment]
Concept ACK for outgoing.

Concept NACK for incoming - I'm worried about giving a false sense of security given the multitude of other ways of doublespending.

**Comment by luke-jr** [comment]
Mostly agreed with @petertodd 

Probably the new icon/colour should be set for *all* unconfirmed transactions signed by a third party, whether incoming or outgoing.

**Comment by paveljanik** [comment]
@petertodd You mean that marking "incoming" tx signalling RBF with this icon can bring false sense of security to non-RBF transactions not marked with this icon? Or?

**Comment by petertodd** [comment]
@paveljanik So the status quo, is that all incoming unconfirmed txs look the same; I'm concerned that showing some types of unconfirmed incoming txs differently than others in UI's intended for general users gives the false impression that Bitcoin Core can reliably tell you about the risk of any particular unconfirmed incoming txs; expert-level UI's like the RPC interface are much less of a concern to me.

**Comment by paveljanik** [comment]
@petertodd In such case, we should not show Unconfirmed at all, add a number input item defaulting to 6 into options and only show transactions with such number of confirmations.

**Comment by paveljanik** [comment]
As all unconfirmed txs are marked with ? in the UI, I think this is safe as long as there is a ? on RBF signalling txs.

**Comment by petertodd** [comment]
@paveljanik But unconfirmed is a state that I think users understand just fine: "Someone intends to send me a transaction, and if they're honest it'll probably confirm (eventually)." There's nothing wrong with showing that.

**Comment by paveljanik** [comment]
Interesting Travis failures - `qInitResources_bitcoin` not found. The log contains:

```
  GEN      qt/qrc_bitcoin.cpp
RCC: Error in 'qt/bitcoin.qrc': Cannot find file 'res/icons/transaction_replaceable.png'
```

**Comment by laanwj** [comment]
Probably you need to add the fine in a Makefile so that it is included in the distribution

**Comment by jonasschnelli** [comment]
Fixed nit and added missing file to Makefile.

**Comment by jonasschnelli** [comment]
@petertodd @luke-jr :
What about keeping the icon but not the orangish color for incoming RBF transactions?

**Comment by gmaxwell** [comment]
@petertodd  ?

I like it with just the icon myself.

**Comment by petertodd** [comment]
@gmaxwell @jonasschnelli Eh, fine, go with that.

**Comment by laanwj** [comment]
utACK 7a30ed6

**Comment by MarcoFalke** [comment]
Looks like getting rid of the orange color is preferred.

@jonasschnelli Mind to take a look?

Also the nit still holds: https://github.com/bitcoin/bitcoin/pull/7817#discussion_r58578763

**Comment by laanwj** [comment]
Removing 0.13 milestone, this missed the feature freeze.

**Comment by btcdrak** [comment]
needs rebase

**Comment by jonasschnelli** [comment]
I think we should first focus on https://github.com/bitcoin/bitcoin/pull/8456 (before we continue with the GUI level).

**Comment by luke-jr** [comment]
@jonasschnelli AFAICT it still has the same problem with just the icon. It gives a false sense of security; there is literally no rational distinction between them for incoming transactions. But maybe I'm missing some use case?

**Comment by paveljanik** [comment]
Rebase needed.

**Comment by jonasschnelli** [comment]
Closing for now due to controversy and inactivity...

**Pull request closed without merging**